aviatrix/resource_aviatrix_transit_external_device_conn.go

```diff
@@ -756,6 +799,20 @@ func resourceAviatrixTransitExternalDeviceConnCreate(d *schema.ResourceData, met
 		break
 	}
 
+	if externalDeviceConn.EnableEdgeUnderlay {
+		re := regexp.MustCompile(`underlay BGP connection (.*) (?:in|on)`)
+		match := re.FindStringSubmatch(result)
+		if len(match) < 2 {
+			return fmt.Errorf("could not get underlay BGP connection name")
+		}
```

Comment:
This looks quite sketchy; we should not have load-bearing human-readable strings. Is there a structured API result we can get this information from?

If we can't get this any other way, what is your plan to defend against changes in the string format?

Reply:
made the backend change to return "conn_name" in API result (separate PR) and updated here to use it directly.